### DISCUSSION TITLE
Why using FFPROBE for duration and not Plex native method?

### DISCUSSION THREAD

**Discussion by codeanator** [CodyBerenson/PGMA-Modernized#116] [Ideas]
Hey,
Saw that you recetly added file duration support but why relying on resource-heavy external component Ffprobe and not the default Plex method for that?
```
fileDuration = 0
for part in media.items[0].parts:
           fileDuration += int(long(getattr(part, 'duration')))
```

**Comment by JPH71** [comment]
@CodeAnator

Thanks for this - I had been looking for an inbuilt plex method but had not
been able to find one....
I will implement this as I hope it will be able to pick up the duration
from films that have split files on disk.
It is always welcome for contributions from members of the group....

Cheers

Jason

**Comment by codeanator** [comment]
U're welc :) I am currently implementing this on my side and it is working. I'll try with split files.
Also, I am trying to turn the cropping method more resilient in case thumbor fails and if Plex not running on Windows. In that case, I would like to fallback to original image even if not optimal. But I do not understand how `def getFilmImages(imageType, imageURL, whRatio)` is working.

This idea would be : 

- Try Thumbor
- Except if nt try local cropping
- Except return original image
- Else if not nt, return original image

**Comment by JPH71** [comment]
There are many python libraries for cropping images... but none of them
seem to work in the plex framework....

As it stands I if the system is not NT, and thumbor fails it will use the
original image as default....

When I get back home, I will look at getFilmImages and let you know what it
is doing.... Alternatively - just paste the code into an email... And I can
give feedback right away....

Cheers

Jason

**Reply by codeanator** [reply]
Where can I find ur email address? 😄

**Comment by JPH71** [comment]
***@***.***

**Reply by codeanator** [reply]
Censored by GH. Of you have a twitter account, follow me and send me a DM at @holedestroyr (NSFW!!)

**Comment by JPH71** [comment]
to get by the censor... you can paste the code into a text file.... or
paste it straight into the email....
Sorry I have been so busy the past two days I forgot to look at the code
when I got home...

**Comment by codeanator** [comment]
It is not working. But don't worry. I fixed it ;) See #117

**Comment by JPH71** [comment]
@CodeAnator
Many thanks for the code in relation to film duration, I have confirmed it
works with stacked films!!!!!
I have also implemented your suggestion in regards to film images.

I am now working on matching the format of the movie names but having issue
with making both the year and segment info optional.... thinking of also
including optional cast names in the filenames at the end of the title...
I have asked a question on StackOverFlow - here is the link:
https://stackoverflow.com/questions/70748753/matching-and-managing-optional-groups-in-regex-python

The file names will be in the following format:
(Studio) - Title (Year) - Segment No [cast1, cast2, cast3]

Studio and Title will not be optional

Kind Regards and thanks for all your help!!!

Jason xxx
